imsim/stamp.py

```diff
@@ -44,9 +44,16 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
             raise galsim.config.SkipThisObject('gal is None (invalid parameters)')
         self.gal = gal
 
-        # Check if the realized flux is 0.
+        # Compute or retrieve the realzed flux.
         self.rng = galsim.config.GetRNG(config, base, logger, "LSST_Silicon")
-        self.realized_flux = galsim.PoissonDeviate(self.rng, mean=gal.flux)()
+        bandpass = base['bandpass']
+        if hasattr(gal, 'flux'):
+            flux = gal.flux
+        else:
+            flux = gal.calculateFlux(bandpass)
```

Comment:
This is the slow thing that we want to cache in skyCatalog.  I guess that will be a different PR, but for now maybe add a TODO comment to remind us that we know this is going to dominate the running time for faint objects.